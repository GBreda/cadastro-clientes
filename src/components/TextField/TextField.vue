<template>
  <label for="text-field" data-test-id="label">{{ label }}</label>
  <input :type="type" name="text-field" data-test-id="input" v-model="value" />
</template>

<script setup>
defineProps({
  label: {
    type: String,
    required: true
  },
  type: {
    type: String,
    default: 'text',
    validator(value) {
      return ['email', 'number', 'tel', 'text'].includes(value);
    }
  }
});

const value = defineModel();
</script>
