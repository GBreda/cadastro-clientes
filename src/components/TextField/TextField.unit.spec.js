import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import TextField from './TextField.vue';

const mountTextField = () => {
  return mount(TextField, {
    props: {
      label: 'Input label'
    }
  });
};

describe('TextField', () => {
  let wrapper;

  it('should mount the component', () => {
    wrapper = mountTextField();

    expect(wrapper.vm).toBeDefined();
  });

  it('should match the snapshot', () => {
    wrapper = mountTextField();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render text label', async () => {
    wrapper = mountTextField();

    await wrapper.setProps({ label: 'Custom label' });

    const label = wrapper.find('[data-test-id="label"]');

    expect(label.html()).toContain('Custom label');
  });

  it('should have the prop type with the value text as default', () => {
    wrapper = mountTextField();

    const input = wrapper.find('[data-test-id="input"]');

    expect(input.attributes('type')).toBe('text');
  });
});
