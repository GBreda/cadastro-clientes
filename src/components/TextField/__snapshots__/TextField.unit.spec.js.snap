// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextField > should match the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <label
    data-test-id="label"
    for="text-field"
  >
    Input label
  </label>
  <input
    data-test-id="input"
    name="text-field"
    type="text"
  />
  
</div>
`;
